src/lmic/lmic.c

```diff
@@ -711,7 +711,7 @@ static CONST_TABLE(u1_t, macCmdSize)[] = {
 static u1_t getMacCmdSize(u1_t macCmd) {
     if (macCmd < 2)
         return 0;
-    if (macCmd >= LENOF_TABLE(macCmdSize) - 2)
+    if ((macCmd - 2) >= LENOF_TABLE(macCmdSize))
```

Comment:
Ugh. Sorry about this one.

Reply:
No problem, but I'm wondering why 0 is returned in the error case, because in `scan_mac_cmds` I see no check except for line 887, which doesn't do what the comment says.

https://github.com/sualko/arduino-lmic/blob/cabf6fbcd4390fcc42102a47c0323b7dafc6723a/src/lmic/lmic.c#L887

Reply:
Yes, that's a bug too -- no test cases for bad mac commands, even in certification.

Should be `if (cmdlen == 0 || cmdlen >> olen - oidx)`.

Reply:
This was fixed in 06b53c79, merged in #555 (and possibly before that, but that's the most recent change).